src/cmd/packages.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 The ChromiumOS Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+use anyhow::anyhow;
+use anyhow::Result;
+use argh::FromArgs;
+use glob::Pattern;
+use lium::cache::KvCache;
+use lium::chroot::Chroot;
+use lium::repo::get_repo_dir;
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// Tast test wrapper
+#[argh(subcommand, name = "packages")]
+pub struct Args {
+    #[argh(subcommand)]
+    nested: SubCommand,
+}
+
+pub static PACKAGE_CACHE: KvCache<Vec<String>> = KvCache::new("package_cache");
+static DEFAULT_BOARD: &str = "host";
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand)]
+enum SubCommand {
+    List(ArgsList),
+}
+pub fn run(args: &Args) -> Result<()> {
+    match &args.nested {
+        SubCommand::List(args) => run_packages_list(args),
+    }
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// Get package list for the target board
+#[argh(subcommand, name = "list")]
+pub struct ArgsList {
+    /// target cros repo directory
+    #[argh(option)]
+    repo: Option<String>,
+
+    /// target board (default: host)
+    #[argh(option)]
+    board: Option<String>,
+
+    /// glob pattern of the packages
+    #[argh(positional)]
+    packages: Option<String>,
+}
+
+fn print_cached_packages(filter: &Pattern, board: &str) -> Result<()> {
+    if let Ok(Some(packages)) = PACKAGE_CACHE.get(board) {
+        for t in &packages {
+            if filter.matches(t) {
+                println!("{t}");
+            }
+        }
+        return Ok(());
+    }
+    Err(anyhow!("No cache found"))
+}
+
+fn update_cached_packages(repodir: &str, board: &str) -> Result<()> {
+    let boardopt = if board == "host" {
+        "--host".to_string()
+    } else {
+        format!("--build-target={}", board)
+    };
+    let chroot = Chroot::new(repodir)?;
+    let list = chroot.exec_in_chroot(&["cros", "workon", &boardopt, "list", "--all"])?;
+    let packages: Vec<String> = list.lines().map(|s| s.to_string()).collect::<Vec<_>>();
+    PACKAGE_CACHE.set(board, packages)?;
+    Ok(())
+}
+
+fn run_packages_list(args: &ArgsList) -> Result<()> {
+    let filter = if let Some(_packages) = &args.packages {
+        Pattern::new(_packages)?
```

Comment:
actually, rust allows variable shadowing, so you can just replace `_packages` with `packages` and it's a recommended way to unwrap the variables like this case.

Reply:
I should have been clearer on this. I just wanted to honor the original code's intention as much as possible.

There is no variable to shadow in this case, right? So what I meant to say was that if the original code's intention was to avoid using `packages` for some reason, this should be done in ways other than preprending an underscore. I would expect a variable name with a prefix of `_` to be not used later on. 

Btw, I'm not going against shadowing. There are cases that makes sense to go for it.

So yeah, I'm good with anything (inc. `packages`) that makes sense for you but not ones starting with an underscore.

Reply:
I found that I can use `unwrap_or()` for this pattern. (that can avoid shadowing discussion :) )